### PR TITLE
Improvements and fixes to the API documentation

### PR DESCRIPTION
**Pull request by xmo-odoo**
Somewhat related to odoo/odoo#74604  as these are issues I found while fixing the other one (and the first commit is quite literally support for a change in odoo/odoo#74604).

### CODE REVIEW

**Comment by xmo-odoo** [comment]
Also hit a weird issue but didn't manage to debug it and ended up ignoring it: since 2.4 `py:attribute` is supposed to [have a `type` field](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#directive-py-attribute).

It works fine when creating a new sphix project, but in the odoo doc project it fails and weirdly it seems to fail *during parsing* (the content which reaches the translator is not correct) but I didn't find any place where we'd override the parsing, only the translation?

**Comment by xmo-odoo** [comment]
Also the CI says it doesn't find "build succeeded" in the log file, but when looking at the logfile it's right there O_o

**Comment by Feyensv** [comment]
It doesn't like the "4 warnings" part :P

**Comment by xmo-odoo** [comment]
> It doesn't like the "4 warnings" part :P

Oh I'd missed that there were actual warnings during the build, I only saw the ones at the top and figured they had nothing to do with me.

It's probably unhappy because those are docstrings I fixed in odoo/odoo#74604 and it always (I assume) builds based on the current master.

File: content/developer/reference/backend/orm.rst [hunk 1]
```diff
@@ -445,44 +446,50 @@ A few field names are reserved for pre-defined behaviors beyond that of
...
-  necessary.
+   default value for :attr:`~odoo.models.BaseModel._rec_name`, used to
+   display records in context where a representative "naming" is
+   necessary.
 
-  :class:`~odoo.fields.Char`
+   :class:`~odoo.fields.Char`
 
-.. attribute:: active
+.. data:: Mode.active
```
Comment (AntoineVDV) [inline content/developer/reference/backend/orm.rst:457]: L

**Review by AntoineVDV — APPROVED** [review]
LGTM (except for the typo).

I'm flagging the PR as a draft until the odoo/odoo counterpart to ease the follow-up.

@robodoo delegate+

**Comment by xmo-odoo** [comment]
@robodoo r+

**Comment by xmo-odoo** [comment]
@robodoo r+

**Comment by xmo-odoo** [comment]
@robodoo rebase-ff

**Pull request closed without merging**